### PR TITLE
MOD-12726 unique snapshot name + new output params for nighly event

### PR DESCRIPTION
**Pull request by AvivDavid23**
1. Make snapshot upload artifact name unique
2. Adding two output parameters to nightly event(snapshot name and version number)

---

> [!NOTE]
> **Medium Risk**
> Medium risk because it changes CI outputs and snapshot/package naming conventions, which can break downstream automation that expects previous artifact paths or filenames.
> 
> **Overview**
> Nightly GitHub Actions now generates and exports additional build metadata: `module-version` (parsed from `src/version.h`) and a unique `snapshot-template` based on branch, timestamp, and run number, and writes both into the run summary.
> 
> Packaging updates in `sbin/pack.sh` append a beta-derived suffix to the snapshot `BRANCH` when `BETA_VERSION` is set, making snapshot artifact names unique across nightly runs while preserving the existing release naming.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 8b5f7b2b50d7020ffc4358f8484f9f6e47940b7e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: sbin/pack.sh [hunk 1]
```diff
@@ -308,6 +308,11 @@ if [[ $WITH_GITSHA == 1 ]]; then
 	BRANCH="${BRANCH}-${GIT_COMMIT}"
 fi
 
+if [[ -n $BETA_VERSION ]]; then
+	BETA_SUFFIX=$(echo "$BETA_VERSION" | cut -d'.' -f4,5,6)
+	BRANCH="${BRANCH}.${BETA_SUFFIX}"
+fi
```
[the remark this answers is not in the document]
Comment (AvivDavid23) [reply sbin/pack.sh:314]: The only scenario that produces the double suffix is: detached HEAD + no BRANCH set + BETA_VERSION set. That's a combination that doesn't happen in the nightly workflow (or any current workflow). So it's a theoretical edge case, not a practical one.

We alsyas call this from GHA, so its fine

File: .github/workflows/flow-linux.yml [hunk 2]
```diff
@@ -158,42 +158,42 @@ jobs:
...
-          name: test-artifacts-${{ inputs.arch }}-${{ matrix.platform.os }}${{ inputs.run_valgrind && '-valgrind' || '' }}${{ inputs.run_sanitizer && '-sanitizer' || '' }}
-          path: |
-            tests/**/*.log
+      # - name: Upload test artifacts
+      #   if: failure()
+      #   uses: actions/upload-artifact@v4
+      #   with:
+      #     name: test-artifacts-${{ inputs.arch }}-${{ matrix.platform.os }}${{ inputs.run_valgrind && '-valgrind' || '' }}${{ inputs.run_sanitizer && '-sanitizer' || '' }}
+      #     path: |
+      #       tests/**/*.log
```
[the remark this answers is not in the document]
Comment (AvivDavid23) [reply .github/workflows/flow-linux.yml]: read the commit message please :)

**Review by LiranAbir — APPROVED** [review]
